### ISSUE TITLE
Publish-Module fails when ModuleVersion isn't in X.Y.Z format

### ISSUE THREAD

**Issue by godefroi** [PowerShell/PowerShellGetv2#88]
Publish-Module fails when ModuleVersion (in the psd1 file) isn't in X.Y.Z format, and the error message is completely unhelpful (`File does not exist`). The original version of PowerShellGet (1.0.0.1) fails, and the latest release on the PS gallery (1.1.2.0) fails as well, in exactly the same way (though the line number is different).

```
myuser@ltp-myuser10 ~\MyModule
$ publish-module -path . -repository MyPSRepo -credential $cred -nugetapikey 'myapikey' -force -verbose
VERBOSE: Repository details, Name = 'MyPSRepo', Location = 'https://myserver/myrepos'; IsTrusted = 'True'; IsRegistered = 'True'.
VERBOSE: Repository details, Name = 'MyPSRepo', Location = 'https://myserver/myrepos'; IsTrusted = 'True'; IsRegistered = 'True'.
VERBOSE: Publish Location:'https://myserver/myrepos'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\MyModule.psm1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\FirstFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\SecondFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\ThirdFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\FourthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\FifthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\SixthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\SeventhFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\EighthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\MyModule\NinthFunction.ps1'.
VERBOSE: Module 'MyModule' was found in 'C:\Users\myuser\MyModule'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\MyModule.psm1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\FirstFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\SecondFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\ThirdFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\FourthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\FifthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\SixthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\SeventhFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\EighthFunction.ps1'.
VERBOSE: Loading module from path 'C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\NinthFunction.ps1'.
VERBOSE: Repository details, Name = 'MyPSRepo', Location = 'https://myserver/myrepos'; IsTrusted = 'True'; IsRegistered = 'True'.
VERBOSE: Using the provider 'PowerShellGet' for searching packages.
VERBOSE: Using the specified source names : 'MyPSRepo'.
VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
VERBOSE: The specified Location is 'https://myserver/myrepos' and PackageManagementProvider is 'NuGet'.
VERBOSE: Searching repository 'https://myserver/myrepos/FindPackagesById()?id='MyModule'' for ''.
VERBOSE: Total package yield:'0' for the specified package 'MyModule'.
Publish-PSArtifactUtility : Failed to publish module 'MyModule': 'File does not exist
(C:\Users\myuser\AppData\Local\Temp\1369426169\MyModule\MyModule.1.0.nupkg).
'.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.1.2.0\PSModule.psm1:1227 char:17
+                 Publish-PSArtifactUtility -PSModuleInfo $moduleInfo `
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : FailedToPublishTheModule,Publish-PSArtifactUtility
```

## Expected Behavior
The publish either succeeds, or fails with a helpful error message

## Current Behavior
The publish fails with a completely unhelpful error message

## Your Environment
```PowerShell

Name                           Value
----                           -----
PSVersion                      5.1.14393.693
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.693
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

ModuleType Version    Name              
---------- -------    ----              
Script     1.1.1.0    PackageManagement 
Binary     1.0.0.1    PackageManagement 
Binary     1.0.0.0    PackageManagement 
Script     1.1.2.0    PowerShellGet     
Script     1.0.0.1    PowerShellGet     

Name                     Version   
----                     -------   
msi                      3.0.0.0   
msu                      3.0.0.0
NuGet                    2.8.5.208 
NuGet                    2.8.5.127 
PowerShellGet            1.1.2.0   
PowerShellGet            1.0.0.1
Programs                 3.0.0.0   
```

**Comment by bmanikm** [comment]
@godefroi 
What value did you specify as ModuleVersion in the module manifest to repro this issue? Have you used the New-ModuleManifest or manually created it?
How did you bootstrap the NuGet.exe?

**Comment by godefroi** [comment]
The ModuleVersion was originally '1.0', which was what was automatically created for me when I ran `New-ModuleManifest`. I changed it to '1.0.0', and the issue went away. Changing the version number to '1' caused the command to fail with a different, only-slightly-less-unhelpful error message.

I didn't manually do anything to "bootstrap" the "NuGet.exe", unless you count downloading it and putting it in my PATH. That was done years ago, as I use that tool often.

**Comment by bmanikm** [comment]
@godefroi I am unable to repro it. Could you please provide the consistent repro steps?

Also please provide the output of following script.

```powershell
if (Test-Path "$env:ProgramData\Microsoft\Windows\PowerShell\PowerShellGet\NuGet.exe") { & "$env:ProgramData\Microsoft\Windows\PowerShell\PowerShellGet\NuGet.exe" ? }

if (Test-Path "$env:LOCALAPPDATA\Microsoft\Windows\PowerShell\PowerShellGet\NuGet.exe") { & "$env:LOCALAPPDATA\Microsoft\Windows\PowerShell\PowerShellGet\NuGet.exe" ? }

Get-Command NuGet | % {& $_.Source ?}
```

**Comment by godefroi** [comment]
My NuGet.exe is in C:\Users\myuser\Utils\NuGet.exe and is version 3.5.0.1996. Neither of your first two commands returned any output.

I checked again, and if the ModuleVersion is `1.0` it fails, and if ModuleVersion is `1.0.0`, it succeeds.

```
$ nuget ?
NuGet Version: 3.5.0.1996
usage: NuGet <command> [args] [options]
Type 'NuGet help <command>' for help on a specific command.

Available commands:

 add         Adds the given package to a hierarchical source. http sources are not supported. For more info, goto https
             ://docs.nuget.org/consume/command-line-reference#add-command.

 config      Gets or sets NuGet config values.

 delete      Deletes a package from the server.

 help (?)    Displays general help information and help information about other commands.

 init        Adds all the packages from the <srcPackageSourcePath> to the hierarchical <destPackageSourcePath>. http fe
             eds are not supported. For more info, goto https://docs.nuget.org/consume/command-line-reference#init-comm
             and.

 install     Installs a package using the specified sources. If no sources are specified, all sources defined in the Nu
             Get configuration file are used. If the configuration file specifies no sources, uses the default NuGet fe
             ed.

 list        Displays a list of packages from a given source. If no sources are specified, all sources defined in %AppD
             ata%\NuGet\NuGet.config are used. If NuGet.config specifies no sources, uses the default NuGet feed.

 locals      Clears or lists local NuGet resources such as http requests cache, packages cache or machine-wide global p
             ackages folder.

 pack        Creates a NuGet package based on the specified nuspec or project file.

 push        Pushes a package to the server and publishes it.
             NuGet's default configuration is obtained by loading %AppData%\NuGet\NuGet.config, then loading any nuget.
             config or .nuget\nuget.config starting from root of drive and ending in current directory.

 restore     Restores NuGet packages.

 setApiKey   Saves an API key for a given server URL. When no URL is provided API key is saved for the NuGet gallery.

 sources     Provides the ability to manage list of sources located in %AppData%\NuGet\NuGet.config

 spec        Generates a nuspec for a new package. If this command is run in the same folder as a project file (.csproj
             , .vbproj, .fsproj), it will create a tokenized nuspec file.

 update      Update packages to latest available versions. This command also updates NuGet.exe itself.

For more information, visit http://docs.nuget.org/docs/reference/command-line-reference
```

**Comment by godefroi** [comment]
This script is just about the cleanest reproduction I can manage:

```
invoke-webrequest https://github.com/anpur/powershellget-module/archive/master.zip -outfile master.zip

expand-archive .\master.zip

$rpath = (md trepo).fullname
$mfile = '.\master\powershellget-module-master\MyModule\MyModule.psd1'

set-content -path $mfile -path (get-content $mfile -raw).replace("'1.0.0'","'1.0'")

Register-PSRepository -Name trepo -SourceLocation $rpath -PublishLocation $rpath -InstallationPolicy Trusted

publish-module -path .\master\powershellget-module-master\MyModule -repository trepo -nugetapikey 'none'
```

**Comment by alexdresko** [comment]
I experienced the same issue today. Using '1.0.0' as the version instead of '1.0' fixed the problem. My manifest was created with `New-ModuleManifest`.

**Comment by bmanikm** [comment]
@alexdresko It looks like you are also using 3.5 version of NuGet.exe. If you need to publish a module with only two parts (1.0) in module version, download NuGet.exe from https://www.nuget.org/NuGet.exe.

**Comment by alexdresko** [comment]
@bmanikm Correct. It's not clear in this thread that nuget v2.8.60717.93 is required to make a 2-part version work, so I hope this comment helps someone. 

I personally do not need a 2-part version number, so it wasn't a problem to make it a 3-part version number. 

Is there a plan to update `publish-module` to support the newer nuget.exe?

**Comment by iricigor** [comment]
Just got the same issue while publishing from **AppVeyor**. I think error details can be visible [here](https://ci.appveyor.com/project/iricigor/easyazurefunction/build/0.3.35) (scroll to bottom). It failed to publish version 0.3. It was resolved by changing version to 0.3.0.

**Comment by bmanikm** [comment]
This issue got resolved with PR #196. Please try with [the latest version of PowerShellGet]( https://www.powershellgallery.com/packages/PowerShellGet) using steps mentioned [here](https://docs.microsoft.com/en-us/powershell/gallery/installing-psget#get-the-latest-version-from-powershell-gallery).
